drivers/display/ls0xx.c

```diff
@@ -0,0 +1,501 @@
+ * -> VCOM is toggled through serial input by software by default
+ *      CONFIG_LS0XX_VCOM_EXTERNAL=n
+ *      EXTMODE pin is connected to VSS
+ *      Important: User has to make sure .write is called periodically
+ *      for toggling VCOM. If there is no data to update then buf can
+ *      be set to NULL then only VCOM will be toggled.
+ * -> To control VCOM using external signal EXTCOMIN, set
+ *      CONFIG_LS0XX_VCOM_EXTERNAL=y
+ *      EXTMODE pin is connected to VDD
+ *      Important: Driver will start a thread which will
+ *      toggle the EXTCOMIN pin every 500ms. There is no
+ *      dependency on user.
+ * -> line_buf format for each row.
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+
+#define LS0XX_SCS_PIN             DT_INST_GPIO_PIN(0, scs_gpios)
+#define LS0XX_SCS_CNTRL           DT_INST_GPIO_LABEL(0, scs_gpios)
+
+#if DT_INST_NODE_HAS_PROP(0, disp_gpios)
+	#define LS0XX_DISP_PIN        DT_INST_GPIO_PIN(0, disp_gpios)
+	#define LS0XX_DISP_CNTRL      DT_INST_GPIO_LABEL(0, disp_gpios)
+#endif
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER)
+	#if defined(CONFIG_LS0XX_VCOM_EXTERNAL) && DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+		#define LS0XX_EXTCOMIN_PIN    DT_INST_GPIO_PIN(0, extcomin_gpios)
+		#define LS0XX_EXTCOMIN_CNTRL  DT_INST_GPIO_LABEL(0, extcomin_gpios)
+	#endif
+#endif
+
+#define LS0XX_PANEL_WIDTH         DT_INST_PROP(0, width)
+#define LS0XX_PANEL_HEIGHT        DT_INST_PROP(0, height)
+
+#define LS0XX_PIXELS_PER_BYTE     8U
+/* Adding 2 for the line number and dummy byte */
+#define LS0XX_BYTES_PER_LINE      ((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE)+2)
+
+#define LS0XX_BIT_WRITECMD        0x01
+#define LS0XX_BIT_VCOM            0x02
+#define LS0XX_BIT_CLEAR           0x04
+
+static uint8_t line_buf[LS0XX_BYTES_PER_LINE];
+
+static struct spi_buf tx_buf;
+static struct spi_buf_set buf_set = {.buffers = &tx_buf, .count = 1};
+
+struct ls0xx_data {
+	struct device *scs;
+#if defined(LS0XX_DISP_CNTRL)
+	struct device *disp;
+#endif
+#if defined(LS0XX_EXTCOMIN_CNTRL)
+	struct device *extcomin;
+#endif
+	struct device *spi_dev;
+	struct spi_config spi_config;
+};
+static uint8_t extcomin_bit;
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER)
+
+#if defined(CONFIG_LS0XX_VCOM_EXTERNAL)
+#if !defined(LS0XX_EXTCOMIN_CNTRL)
+#error "CONFIG_LS0XX_VCOM_EXTERNAL is 'y' but \
+extcomin-gpios is not present in devicetree"
+#endif /* LS0XX_EXTCOMIN_CNTRL */
+#define LS0XX_EXTCOMIN_BIT 0x00
+#else
+#define LS0XX_EXTCOMIN_BIT extcomin_bit
+#endif /* CONFIG_LS0XX_VCOM_EXTERNAL */
+
+#if defined(CONFIG_LS0XX_VCOM_EXTERNAL)
+/**
+ * @brief Toggle VCOM to avoid DC bias
+ *
+ * @details This API will toggle either the EXTCOMIN pin
+ *          or the M1 bit in serial comm to toggle VCOM
+ *
+ * @param driver Pointer to the display driver
+ */
+static void ls0xx_vcom_toggle(void *a, void *b, void *c)
```

Comment:
Why can not you use a workqueue thread and trigger it from a timer?

Reply:
we could. but i thought it would be complicated for a simple driver like this.